src/workers/filesystems/infrastructure/SyncRemoteFileSystem.ts

```diff
@@ -0,0 +1,666 @@
+        // see: https://anchortagdev.com/range-error-maximum-call-stack-size-exceeded-error-using-spread-operator-in-node-js-javascript/
+
+        for (const file of batch.files) files.push(file);
+
+        for (const folder of batch.folders) folders.push(folder);
+
+        thereIsMore = batch.folders.length === PAGE_SIZE;
+
+        if (thereIsMore) offset += PAGE_SIZE;
+      } catch (err) {
+        await handleFetchError(err, 'Fetching tree', `offset: ${offset}`);
+      }
+    }
+
+    return { files, folders };
+  }
+
+  async function handleFetchError(
+    err: any,
+    action: string,
+    additionalInfo?: string
+  ) {
+    if (err instanceof ProcessError) throw err;
+
+    const details = createErrorDetails(err, action, additionalInfo);
+
+    if (await isOnline()) {
+      throw new ProcessError('NO_REMOTE_CONNECTION', details);
+    } else {
+      throw new ProcessError('NO_INTERNET', details);
+    }
+  }
+
+  return {
+    kind: 'REMOTE',
+
+    async getCurrentListing() {
+      const tree = await getTree();
+
+      const listing: RemoteListing = {};
+      const readingMetaErrors: Array<ReadingMetaErrorEntry> = [];
+
+      traverse(baseFolderId);
+
+      function traverse(currentId: number, currentName = '') {
+        const filesInThisFolder = tree.files.filter(
+          (file) => file.folderId === currentId
+        );
+        const foldersInThisFolder = tree.folders.filter(
+          (folder) => folder.parent_id === currentId
+        );
+
+        filesInThisFolder
+          .map((file) => ({
+            name:
+              currentName +
+              crypt.decryptName(
+                file.name,
+                file.folderId.toString(),
+                file.encrypt_version
+              ) +
+              (file.type ? `.${file.type}` : ''),
+            file,
+          }))
+          .filter(({ name }) => {
+            const isValid = fileNameIsValid(name);
+
+            if (!isValid) {
+              Logger.warn(
+                `REMOTE file with name ${name} will be ignored due an invalid name`
+              );
+              return false;
+            }
+
+            return true;
+          })
+          .forEach(({ file, name }) => {
+            const modificationTime = getSecondsFromDateString(
+              file.modificationTime
+            );
+            listing[name] = RemoteItemMetaData.from({
+              modtime: modificationTime,
+              size: file.size,
+              isFolder: false,
+              id: file.id,
+            });
+            cache[name] = {
+              id: file.id,
+              parentId: file.folderId,
+              isFolder: false,
+              bucket: file.bucket,
+              fileId: file.fileId,
+              modificationTime,
+              size: file.size,
+            };
+          });
+
+        foldersInThisFolder.forEach((folder) => {
+          const name =
+            currentName +
+            crypt.decryptName(
```

Comment:
Worth to start using `plainName` if the field exists here. Therefore we will avoid having to publish a new Desktop App to remove the decrypt-name-always assumption.

Reply:
Created a pr to return the plain name on drive-server https://github.com/internxt/drive-server/pull/385